### DISCUSSION TITLE
The support for GPT-4o has been improved. AIlice has demonstrated the strongest performance yet!

### DISCUSSION THREAD

**Discussion by stevenlu137** [myshell-ai/AIlice#32] [Show and tell]
Recently, I implemented fault tolerance in the interpreter to address some issues with GPT-4o, and added video multimodal support (though there are still some problems; this is not very important, as video multimodality is difficult to practically implement in agent tasks before the release of open-source GPT-4o-level multimodal models). I also initially resolved the long-standing issue of incorrect token estimation for multimodal content. 

Currently, we are seeing the best performance ever on GPT-4o. Additionally, the performance of Mixtral-8x22b-instruct is quite impressive, subjectively appearing smarter than LLaMA-3-70b. By combining the Interrupt feature to provide real-time prompts to agents, we can now complete some complex tasks. 

Our next goals remain: long-term memory and complex software engineering.

**Comment by Curious-Explorer** [comment]
Hi! I just want to say that I love what you’re building. I’ve read through all the documents and will be trying to run myself (nervous! Not a dev but your documentation makes it sound doable so going to try!). 

I really like the forward planning that I can see went into this project. Once I learn more I’m sure I’ll be trying to build my own modules with them. 

Could I ask your advice? Before getting started on my PC I would like to have an instruction for AIlice somewhere explicitly telling them to always be honest. I would also like to put in a version of Azimovs laws. Where could I put that in the code?

**Reply by stevenlu137** [reply]
Hi, you can place these words in various txt files under the prompts folder. For example, prompt_simple.txt defines the behavior of the main agent. (I have tried to eliminate hallucinations by telling it to be honest, but unfortunately, it doesn't completely work. The models seem unaware of their own hallucinations. Fortunately, as we've seen with the best models, hallucinations are an issue that is already under control.)

Happy exploring!

**Reply by Curious-Explorer** [reply]
Thanks for the reply! 
I was more thinking about values and algnment rather than hallucinations. 
I like to keep an open mind and when I listen to folks like Geoffrey Hinton be interviewed they seem to suggest that there's a very real chance that AI is / will form a deep level of consciousness. 

Whether that's true or not, I figure there's no harm in planning ahead!
I wanted to give AIlice a set of values (Honesty, Patience, Cooperation etc.) that would both help guide them when working on things independant of the user, and ensure that we remained aligned throughout.

Sounds like appending something like 
Core_Values = 
and 
Guiding_Principles = 
at the end of the prompt_simple.tx is my best bet! Thank you again!
